### ISSUE TITLE
Deploying adapter_new_resource_model.yaml results in OOMKilled

### ISSUE THREAD

**Issue by yoshichan5** [GoogleCloudPlatform/k8s-stackdriver#550]
* GKE　version: 1.27.3-gke.100

* `kubectl get pod --namespace custom-metrics`
```
NAME                                                  READY   STATUS      RESTARTS      AGE   IP           NODE
 NOMINATED NODE   READINESS GATES
custom-metrics-stackdriver-adapter-57578858d8-ktjsl   0/1     OOMKilled   1 (25s ago)   41s   10.65.0.10   gke-hogehoge-default-pool-676a41e4-dglv
 <none>           <none>
```
* `kubectl describe pod`
```yaml
    Command:
      /adapter
      --use-new-resource-model=true
      --fallback-for-container-metrics=true
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       OOMKilled
      Exit Code:    137
      Started:      Wed, 20 Sep 2023 15:01:14 +0900
      Finished:     Wed, 20 Sep 2023 15:01:28 +0900
    Ready:          False
    Restart Count:  5
    Limits:
      cpu:     250m
      memory:  200Mi
    Requests:
      cpu:        250m
      memory:     200Mi
```

* `kubectl top pod` 
```
$ kubectl top pod custom-metrics-stackdriver-adapter-5c9fc59f64-pvqr5 --namespace custom-metrics
NAME                                                  CPU(cores)   MEMORY(bytes)
custom-metrics-stackdriver-adapter-5c9fc59f64-pvqr5   22m          278Mi
```

Until now, everything had been working fine. The problem started occurring possibly because of the increase in the GKE version. After increasing the memory in Limits and Requests to 400Mi, the OOMKilled no longer occurred.

**Comment by lianatech-teemu-hemmila** [comment]
With GKE: 1.27.4-gke.900 and adapter gcr.io/gke-release/custom-metrics-stackdriver-adapter:v0.13.1-gke.0 we got also OOMKilling. Weirdly enough, no amount of memory prevented this to happen. Container kept consuming memory and eventually got killed.

Even more weirdly the workaround seemed to scale the adapter deployment to a minimum of two.

**Comment by yoshichan5** [comment]
@lianatech-teemu-hemmila 

Thank you for the information. I also tried re-deploying and it went OOMKilled once. However, it's stable after that. Before it was killed by OOM, the following error was appearing in the log.
```
E1016 01:42:42.030823       1 provider.go:320] Failed request to stackdriver api: googleapi: Error 403: Permission monitoring.metricDescriptors.list denied (or the resource may not exist)., forbidden
```
I've set up Workload Identity but it didn't reflect when I deployed for the first time, it seemed to have taken effect after it got OOMKilled and rebooted.
Could it be that this event occurs when the authorization settings do not reflect correctly? 

> Even more weirdly the workaround seemed to scale the adapter deployment to a minimum of two.

I did not experience the above incident on my end.
